hub/core/transform/transform_tensor.py

```diff
@@ -59,7 +59,7 @@ def __getitem__(self, index):
         return TransformTensor(
             name=self.name,
             dataset=self.dataset,
-            base_tensor=self.base_tensor,
+            base_tensor=self._base_tensor,
```

Comment:
use self.base_tensor here

Reply:
if I do so, it leaks

Reply:
If you don't base tensor will always be None

Reply:
but this is wrong, we can't pass None here

Reply:
this can't be the fix then

Reply:
we need to have a reference to base tensor

Reply:
I see, I will create a new class for base tensor then, in order not to have a circular reference

Reply:
I don't see how that'll fix the issue?

Reply:
I think we need to get rid of recursion